onnxruntime/python/tools/transformers/fusion_attention.py

```diff
@@ -351,24 +404,23 @@ def fuse(self, normalize_node, input_name_to_nodes, output_name_to_node):
             logger.debug("fuse_attention: failed to match mask path")
             return
 
-        if matmul_v.input[0] == root_input and matmul_q.input[0] == root_input and matmul_v.input[0] == root_input:
+        if matmul_v.input[0] == root_input and matmul_q.input[0] == root_input and matmul_k.input[0] == root_input:
             mask_index = self.attention_mask.process_mask(mask_nodes[-1].input[0])
 
             attention_last_node = reshape_qkv if einsum_node is None else transpose_qkv
 
-            num_heads, hidden_size = self.get_num_heads_and_hidden_size(reshape_q)
-            if num_heads <= 0 or hidden_size <= 0 or (hidden_size % num_heads) != 0:
-                logger.debug("fuse_attention: failed to detect num_heads or hidden_size")
-                return
-
+            # number of heads are same for all the paths, hence to create attention node, we pass the self.num_heads
+            # the hidden_size represents the hidden input size, this is used as needed but hidden sizes for Q, K are extracted appropriately
             new_node = self.create_attention_node(mask_index, matmul_q, matmul_k, matmul_v, add_q, add_k, add_v,
-                                                  num_heads, hidden_size, root_input, attention_last_node.output[0])
+                                                  self.num_heads, self.hidden_size, root_input, attention_last_node.output[0])
```

Comment:
Any reason to use self.num_heads instead of the the one from get_num_heads_and_hidden_size(reshape_q)?

Reply:
Updated to preserve the current behavior and use user input in case input_hidden_size is different from q hidden dim.